app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,68 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.all
+  end
+
+  def login
+    id = params.permit(:login_id)[:login_id].to_i
+
+    if (id <= Market.last.id && id > 0)
+      redirect_to "/markets/#{ id }/dashboard"
+    else
+      redirect_to "/markets/market_not_found"
```

Comment:
Also, using `render` lets you give your users another chance to fix their mistake immediately instead of making them navigate back to wherever they were before the error happened.